docs/kb/semgrep-ci/azure-using-templates-with-semgrep.md

```diff
@@ -8,9 +8,11 @@ description: Running Semgrep commands in Azure Pipelines templates.
 
 ## Motivation
 
-Complex CI configs can take a lot of work to maintain and modify. It can therefore be a good practice to encapsulate some functionalities into external templates. By referencing these external templates, you are able to reuse functionalities more concisely and without having to define them again.
+Complex CI configurations can take a lot of work to maintain and modify. Creating external configurations that encapsulate a chunk of configuration logic and referencing them keeps configurations concise and consistent.
 
-The primary goal of Azure templates is to extract some functionalities defined in large YAML config files and place them into their own separate [templates](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops). This improves a pipeline's readability and maintainability. This guide explains how to:
+Azure templates extract configuration logic that was previously defined in large YAML configuration files and place it into its own [template](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) referenced in the main YAML file. This improves a pipeline's readability and maintainability. 
```

Comment:
In the first sentence:

> Azure templates extract configuration logic that was previously defined in large YAML configuration files and place it into **its** own

I think the "its" in bold is referring to Azure templates, which is plural, so it should be "their"?

Reply:
I meant it to refer to "configuration logic", but that definitely tells me this sentence is still confusing (I've tried to rewrite it a few times during this update 😅 ).

I'm kind of wondering if it's worth re-writing both paragraphs here to something simpler.

Complex CI configurations housed in large YAML files take a lot of work to maintain and modify. Azure [templates](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) extract chunks of logic from larger configurations and encapsulate them in external template files. The template can then be referenced in multiple configurations, keeping pipelines more readable and maintainable.

Reply:
Your approach here, where you open with the problem, is very clear:
> Complex CI configurations housed in large YAML files take a lot of work to maintain and modify. Azure [templates](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops) extract chunks of logic from larger configurations and encapsulate them in external template files. The template can then be referenced in multiple configurations, keeping pipelines more readable and maintainable.

Let's use it.